index.js

```diff
@@ -56,26 +54,46 @@ export function checkDataIntegrityProofFormat({
             'Expected proof to be either an object or an array.'
           );
         });
-        it('`type` field MUST exist and be a string.', function() {
+        it('if "id" field exists, it MUST be a valid URL.', function() {
           this.test.cell = {columnId: vendorName, rowId: this.test.title};
           for(const proof of proofs) {
-            proof.should.have.property('type');
-            proof.type.should.be.a('string');
+            if(proof.id) {
+              let result;
+              let err;
+              try {
+                result = new URL(proof.id);
+              } catch(e) {
+                err = e;
+              }
+              should.not.exist(err, 'Expected URL check of the "proof.id" ' +
+                'to not error.');
+              should.exist(result, 'Expected "proof.id" to be a URL.');
+            }
           }
         });
-        it('`type` field MUST be `DataIntegrityProof`.', function() {
+        it('"type" field MUST exist and be a string.', function() {
           this.test.cell = {columnId: vendorName, rowId: this.test.title};
           for(const proof of proofs) {
             proof.should.have.property('type');
             proof.type.should.be.a(
-              'string',
-              'Expected `proof.type` to be a string.'
-            );
-            proof.type.should.equal('DataIntegrityProof');
+              'string', 'Expected `proof.type` to be a string.');
+          }
+        });
+        it('"type" field MAY be either "DataIntegrityProof" or ' +
+          '"Ed25519Signature2020".', function() {
+          this.test.cell = {columnId: vendorName, rowId: this.test.title};
+          for(const proof of proofs) {
+            proof.should.have.property('type');
+            proof.type.should.be.a('string');
+            const expectedTypes = [
+              'DataIntegrityProof', 'Ed25519Signature2020'
```

Comment:
Yes, this test shouldn't exist here. It should exist in the specific cryptosuite being tested. I'd remove the `Ed25519Signature2020` part of this test and just make it a generic test for any "DataIntegrity" cryptosuite.

Reply:
Removed `Ed25519Signature2020`, now only checks if the type is `DataIntegrityProof` [4b62348](https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/12/commits/4b6234899e3f33b56818dd277dd2e42dbdbe408f).